posthog/utils.py

```diff
@@ -541,3 +541,15 @@ def get_safe_cache(cache_key: str):
         except:
             pass
     return None
+
+
+def is_valid_uuid4(uuid_string: str) -> bool:
+    try:
+        val = uuid.UUID(uuid_string, version=4)
+    except ValueError:
+        return False
+    # If the uuid_string is a valid hex code,
+    # but an invalid uuid4,
+    # the UUID.__init__ will convert it to a
+    # valid uuid4. This is bad for validation purposes.
+    return val.hex == uuid_string
```

Comment:
We tend to use [`UUIDT`](https://github.com/PostHog/posthog/blob/02f611b6988f85cd60afc7bf0996ae51e19d4d9e/posthog/models/utils.py#L10), which doesn't precisely adhere to UUIDv4 spec, so may have to revisit this sometime